src/cmd/environments/create/create-env.go

```diff
@@ -19,27 +20,29 @@ var CreateEnvCmd = &cobra.Command{
 	RunE: func(_ *cobra.Command, args []string) error {
 		ctxTimeout, cancel := context.WithTimeout(context.Background(), config.DefaultTimeout)
 		defer cancel()
-		envsClient := environments.NewClientFromToken(viper.GetString(config.OtterizeAPIAddressKey), config.GetAPIToken(ctxTimeout))
+
+		c := cloudclient.NewClientFromToken(viper.GetString(config.OtterizeAPIAddressKey), config.GetAPIToken(ctxTimeout))
 
 		name := viper.GetString(NameKey)
 		labels := viper.GetStringMapString(LabelsKey)
-
-		var env environments.EnvFields
-		var err error
-
-		if viper.GetBool(ExistsOkKey) {
-			env, err = envsClient.GetOrCreateEnv(ctxTimeout, name, labels)
-		} else {
-			env, err = envsClient.CreateEnv(ctxTimeout, name, labels)
-		}
-
+		labelsInput := lo.Ternary(len(labels) == 0, nil, lo.ToPtr(cloudclient.LabelsToLabelInput(labels)))
+
+		r, err := c.Client.CreateEnvironmentMutationWithResponse(ctxTimeout,
+			cloudapi.CreateEnvironmentMutationJSONRequestBody{
+				Name:   name,
+				Labels: labelsInput,
+			},
+		)
 		if err != nil {
 			return err
 		}
 
-		formatted, err := output.FormatItem(env, func(env environments.EnvFields) string {
-			return fmt.Sprintf("Environment created with ID: %s", env.Id)
-		})
+		if cloudclient.IsErrorStatus(r.StatusCode()) {
+			return output.FormatHTTPError(r)
```

Comment:
I think this should be part of the client itself - the responsibility to recognize that an error occurred should be with the shared client code, instead of with each caller - it shouldn't be possible to forget to handle a http 500

Reply:
the problem here is that some error codes should actually be handled by the caller in some cases (e.g. error 404). the other issue is that almost 100% of the code outside of the cmd implementation is auto-generated, and so we'd have to have some wrappers for it. I'll try to think about a better approach for generalizing it (opening an issue).